src/index.js

```diff
@@ -122,8 +120,16 @@ function Swarm (peerInfo) {
       this.muxedConns[key].muxer.end()
     })
 
-    parallel(Object.keys(this.transports).map((key) => {
-      return (cb) => this.transports[key].close(cb)
+    const transports = this.transports
+
+    parallel(Object.keys(transports).map((key) => {
+      return (cb) => {
+        parallel(transports[key].listeners.map((listener) => {
+          return (cb) => {
+            listener.close(cb)
+          }
+        }), cb)
+      }
```

Comment:
This can be written a little bit more compact

```js
parallel(Object.keys(transports).map((key) => (c) => {
  parallel(transports[key].listeners.map((listener) => listener.close.bind(listener)))
}))
```